### PR TITLE
Navbar improvements

### LINKED ISSUE #989: Show interface before login
Currently the first view when navigating to I-analyzer is the login screen. It would be nice if the home page showed more of the actual interface, with the login menu on the side or such.

### PR DESCRIPTION
**Pull request by lukavdplas**
Some accessibility improvements to the navbar:
- The menu toggle on mobile is now keyboard accessible
- The user dropdown has improved keyboard accessibility (see #794)
- Adds "skip to content link" at the start
- With the exception of logging out, menu items have an `href` link, rather than triggering a script event that sets the route. This makes links more predictable for users.
- Use `<header>` and `<footer>` in the app component so the header/footer get appropriate roles.

I've also cleaned up the logic of the navbar CSS and typescript. There were some arcane bits of logic without an adequate explanation of what they were for, so I *hope* it all works as before.

One final improvement is that the header is now visible when logged out, which also means new users can view the "about" page before registering. (close https://github.com/UUDigitalHumanitieslab/I-analyzer/issues/989 )

### CODE REVIEW

File: frontend/src/app/menu/menu.component.html [hunk 1]
```diff
@@ -1,50 +1,113 @@
...
-                    <span aria-hidden="true"></span>
-                    <span aria-hidden="true"></span>
+            <a  role="button" tabindex="0"
+                (click)="toggleMenu($event)" (keydown.enter)="toggleMenu($event)"
+                class="navbar-burger" [class.is-active]="menuOpen"
+                aria-label="menu" [attr.aria-expanded]="menuOpen"
+                data-target="navigation-menu">
+                <span aria-hidden="true"></span>
+                <span aria-hidden="true"></span>
+                <span aria-hidden="true"></span>
```
Comment (JeltevanBoheemen) [inline frontend/src/app/menu/menu.component.html:23]: Why the repeated span?

File: frontend/src/app/menu/menu.component.ts [hunk 2]
```diff
@@ -1,118 +1,130 @@
...
+
+    toggleDropdown() {
+        this.dropdownOpen$.next(!this.dropdownOpen$.value);
     }
 
-    public gotoAdmin() {
-        window.location.href = environment.adminUrl;
+    toggleMenu() {
+        this.menuOpen = !this.menuOpen;
     }
```
Comment (JeltevanBoheemen) [inline frontend/src/app/menu/menu.component.ts:77]: Inconsistent approach between these two methods.

**Review by JeltevanBoheemen — CHANGES_REQUESTED** [review]
Code looks good and makes sense. 

One important request: I like the visible header when not logged in/registered. But the way back to the login page from the about page is not apparent. Add a nav element to that effect.

Laughed at the description of 'arcane' code, I noticed this when working on the auth service. It all seems to work. Reminder to thoroughly test solis/regular logins AND registration/email confirmation.

File: frontend/src/app/menu/menu.component.html (hunk 1 above)
Comment (lukavdplas) [reply frontend/src/app/menu/menu.component.html:23]: I thought this was a mistake, but this is how the bulma documentation does it. My best guess is that this is how they create the "hamburger icon". I haven't tried removing it though.

**Pull request merged**